rust/nasl-interpreter/src/built_in_functions/misc.rs

```diff
@@ -154,6 +182,128 @@ pub fn gunzip(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, F
+        Some(ContextType::Value(NaslValue::Number(x))) =>  *x as u32,
+        _ => 0,
+    };
+    let mday = match register.named("mday") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as u32,
+        _ => 0,
+    };
+    let mon = match register.named("mon") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as u32,
+        _ => 1,
+    };
+    let year = match register.named("year") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as i32,
+        _ => 0,
+    };
+
+    // TODO: fix isdst
+    let isdst = match register.named("isdst") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as i32,
+        _ => -1,
+    };
+
+    let offset = chrono::Local::now().offset().fix().local_minus_utc();
+    let r_dt = Utc.with_ymd_and_hms(year, mon, mday, hour, min, sec);
+    match r_dt {
+        LocalResult::Single(x) => Ok(NaslValue::Number(x.naive_local().timestamp() - offset as i64)),
+        _ => Ok(NaslValue::Null),
+    }
+}
+
+
+
+/// Returns an dict(mday, mon, min, wday, sec, yday, isdst, year, hour) based on optional given time in seconds and optional flag if utc or not.
+pub fn localtime(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+    let named = register.named("utc");
+    let utc_flag;
+    match named {
+        Some(ContextType::Value(NaslValue::Number(x))) => {utc_flag = *x != 0i64;},
+        Some(ContextType::Value(NaslValue::Boolean(x))) => {utc_flag = *x;},
+        _ => {utc_flag = false;},
+    };
+        
+    let mut tictac_local: chrono::DateTime<Local> = Local::now();
+    let mut tictac_utc: chrono::DateTime<Utc> = Utc::now();
+    let tictac: Vec<&str>;
+    let mut secs: i64 = 0;
+    let positional = register.positional();
+    if !(positional.is_empty()) && positional[0] != NaslValue::Number(0)
+    {
+        match &positional[0] {
+            NaslValue::Number(x) => {
+                if *x > 0 {
+                    secs = *x as i64;
+                }
+                else {
+                    return Err(FunctionError::new("localtime", ("0", "numeric").into()));
+                };
+            },
+            NaslValue::String(x) => {
+                let secstr: Vec<&str> = x.split(".").collect();
+                let r_secs = secstr[0].to_string().parse::<i64>();
+                match r_secs {
+                    Ok(x) => secs = x as i64,
+                    Err(_) => return Err(FunctionError::new("localtime", ("0", "numeric").into())),
+                }
+            },
+            _ => return Err(FunctionError::new("localtime", ("0", "numeric").into())),
+        }
+    }
+
+    let strfmt;
+    if utc_flag {
+        if secs != 0 {
+            match Utc.timestamp_opt(secs, 0) {
+                LocalResult::Single(x) => {tictac_utc = x;},
+                _ => (),
+            };
+        }
+        strfmt = tictac_utc.format("%S %M %H %d %m %Y %w %j").to_string();
```

Comment:
is there no simpler way to just get the fields of a datetime construct than creating a format string, than create a vector tbased on that to get the index instead of just calling the corresponding methods of the date struct? This seems a bit complicated.